docs/specifications/tests/DNSQueryAndResponsSpecification.md

```diff
@@ -0,0 +1,154 @@
+# DNS Query and Respons Specification
+
+**Table of contents**
+* [Overview](#Overview)
+* [Application of this specification](#application-of-this-specification)
+* [Default setting in *DNS Query*](#default-setting-in-dns-query)
+* [Default setting in *EDNS Query*](#default-setting-in-edns-query)
+* [Default setting in *DNSSEC Query*](#default-setting-in-dnssec-query)
+* [Default handling of a *DNS Response*](#default-handling-of-a-dns-response)
+* [Default handling of an *EDNS Response*](#default-handling-of-an-edns-response)
+* [Default handling of a *DNSSEC response*](#default-handling-of-a-dnssec-response)
+
+
+## Overview
+
+Almost all [test cases] emit DNS queries and react on the responses. This
+document defines the default "setting" of a DNS query and the default handling
+of "setting" in the DNS response. The meaning of "setting" here will be clear
+from the specification below.
+
+
+## Application of this specification
+
+This specification applies to all [test case specifications][Test Cases] that has
+an explicit reference to this document.
+
+
+## Default setting in *DNS Query*
+
+A *DNS Query* has the following default setting. A test case specification can
+refer to a *DNS Query* with one or several changes to the setting items.
+Specifically, the test case specification must always specify query name and
+query type.
+
+|Setting Item |Default value |Comment                      |
+|:------------|:----------|:-------------------------------|
+|Protocol     | UDP       |                                |
+|OpCode       | "query"   | Always "query" in a query      |
+|QR flag      | unset     | Always unset in a query        |
+|AA flag      | unset     |                                |
+|TC flag      | unset     |                                |
+|RD flag      | unset     |                                |
+|RA flag      | unset     |                                |
+|AD flag      | unset     |                                |
+|CD flag      | unset     |                                |
+|RCODE        | "NoError" | Always "NoError" in a query    |
+|Query name   | -         | Defined in test case           |
+|Query type   | -         | Defined in test case           |
+|ENDS         | no        | No OPT record is included      |
+
+
+## Default setting in *EDNS Query*
+
+An *EDNS Query* inherit the default setting from a *DNS Query* except for the
+setting items specified below.
+
+A test case specification can refer to an *EDNS Query* with one or several
+changes to the setting items. Specifically, the test case specification must
+always specify query name and query type.
```

Comment:
Does it really make sense to mention query name and query type again here?

There's another one like this below.